superset/views/core.py

```diff
@@ -1796,46 +1795,29 @@ def testconn(self):
             uri = request.json.get('uri')
             db_name = request.json.get('name')
             impersonate_user = request.json.get('impersonate_user')
-            database = None
+
+            # extras is sent as json, but required to be a string in the Database model
+            extra_json = request.json.get('extras', {})
```

Comment:
I would probably just make this one line called `extras` (inline with the JSON name), i.e., 

```
extras = json.dumps(request.json.get('extras', {}))
```